plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Inline.hs

```diff
@@ -223,17 +238,44 @@ We rename both terms and types as both may have binders in them.
 
 -- | Run the inliner on a single non-recursive let binding.
 processSingleBinding
-    :: forall tyname name uni fun a. InliningConstraints tyname name uni fun
-    => Term tyname name uni fun a -- ^ The body of the let binding.
-    -> Binding tyname name uni fun a -- ^ The binding.
-    -> InlineM tyname name uni fun a (Maybe (Binding tyname name uni fun a))
-processSingleBinding body = \case
-    TermBind a s v@(VarDecl _ n _) rhs -> do
-        maybeRhs' <- maybeAddSubst body a s n rhs
-        pure $ TermBind a s v <$> maybeRhs'
-    TypeBind a v@(TyVarDecl _ n _) rhs -> do
+    :: forall tyname name uni fun ann. InliningConstraints tyname name uni fun
+    => Term tyname name uni fun ann -- ^ The body of the let binding.
+    -> Binding tyname name uni fun ann -- ^ The binding.
+    -> InlineM tyname name uni fun ann (Maybe (Binding tyname name uni fun ann))
+processSingleBinding body =
+    -- when we encounter a binding that is a function, we add it to the global map
+    -- `Utils.CalledVarEnv`. When we check the body of the let binding we look in this map for
+    -- call site inlining.
+    let addVarToMap letBody v ann s n rhs = do
+            -- we want to do unconditional inline if possible
+            maybeRhs' <- maybeAddSubst letBody ann s n rhs
```

Comment:
okay, so it's processed.

Reply:
I would not expect this function to also modify the substitution! It sounds like it's just adding the `VarInfo` but it isn't.